### ISSUE TITLE
Discuss the 'sylt-mentaility'

### ISSUE THREAD

**Issue by FredTheDino** [FredTheDino/sylt-lang#76]
TL;DR
We need to make the goals clearer in the README file.

How do we think when making Sylt? We have set up the goals of static typechecking, but do we want to take it further? Can we catch more errors? Do we want to catch these errors? Where to we want to look for answers?

------------------
**WARNING** Everthing bellow here is opinion

I think we want to add in all tools we can, to help people
produce correct programs. And in some ways I think that
requires us to make some "extreme" decisions, referring
to "unused variables are illegal". But I think we also want
something that's a lot more hackable. And the way to do
that might be warnings. Because we aren't making a language
for the `masses`, we're making a language for GameJams.

So from that perspective, I don't think you want to mess
a lot with the language or enable opt-in features. You don't
want complicated build systems or people using different
flags when building and getting different results. This leads
me to having all "warnings" as errors.

Or maybe it's a good idea to have different levels of "errors",
and 3 low-priority errors cause a "failed to compile" error.
Like a 3-strikes system, saying "hey, you need to fix this!".

**Comment by sornas** [comment]
> You don't want complicated build systems or people using different flags when building and getting different results.

Agree. Flags are out of the picture.

> Or maybe it's a good idea to have different levels of "errors", and 3 low-priority errors cause a "failed to compile" error. Like a 3-strikes system, saying "hey, you need to fix this!".

Disagree. Very arbitrary.

After discussion:

The intended use case is during game jams, where people want to Get Things Done:tm: which is why I don't think we want to not run a "logically valid" program just because there's an unused variable. Removing the unused variable might mean removing a lot of lines in a lot of places (which the compiler currently can't help you with, but could in the future).

We also discussed the possibilities of a linter which we both liked. The unused variable detection is the same, however, so we merged #67 as-is and went on to do new things.

**Comment by FredTheDino** [comment]
This should move to the README.md
